code/modules/surgery/organs/organ_external.dm

```diff
@@ -312,7 +312,8 @@ This function completely restores a damaged organ to perfect condition.
 
 	// handle internal organs
 	for(var/obj/item/organ/internal/current_organ in internal_organs)
-		current_organ.rejuvenate()
+		if(!istype(current_organ,/obj/item/organ/internal/cyberimp)|| !istype(current_organ,/obj/item/organ/internal/xenos) || !istype(current_organ,/obj/item/organ/internal/hivelord_core))
```

Comment:
i am trying to get it not to remove those organs...i will need to rethink this now...